### ISSUE TITLE
Search does not load with sphinx-autobuild

### ISSUE THREAD

**Issue by hugoShaka** [jbms/sphinx-immaterial#94]
tl;dr using sphinx-autobuild to serve the site leads to the ressource `searchindex.js` not loading and the search being broken

What happens:
* no base url is specified in config, the url is inferred from getLocation: `http://my-machine:8501/`
  https://github.com/jbms/sphinx-immaterial/blob/230a039dd62adaa79d4b4efd93f23d84b989eebf/src/assets/javascripts/_/index.ts#L106
* user starts searching and this loads the searchindex https://github.com/jbms/sphinx-immaterial/blob/230a039dd62adaa79d4b4efd93f23d84b989eebf/src/assets/javascripts/sphinx_search.ts#L62-L68
* the url is constructed by adding a second `/` at the beginning https://github.com/jbms/sphinx-immaterial/blob/230a039dd62adaa79d4b4efd93f23d84b989eebf/src/assets/javascripts/sphinx_search.ts#L29-L31
* the browser tries to load  `http://my-machine:8501//searchindex.js`
* While this URL sounds RFC-compliant the sphinx-authbuild webserver does not like it and answers 403s

From what I understand, removing the `/` in getAbsoluteUrl would not break anything and solve the issue. I can send a PR if needed.

**Comment by 2bndy5** [comment]
Modifying the TS files is frowned upon here because they are meant to be as close as possible to upstream (mkdocs-material theme). That said, only _sphinx-immaterial/src/assets/javascripts/_/index.ts_ really applies to that stipulation.

1. Have you tried that solution about removing the `/` in `getAbsoluteUrl`? I'm concerned that might lead to a regression.
2. There are other ways to host a webserver that "watches" for changes about a certain path.

**Comment by 2bndy5** [comment]
>I can send a PR if needed.

PRs are welcome. I see that extra `/` when browsing the local path (no webserver), so I can't imagine why that solution would cause a regression.

>While this URL sounds RFC-compliant the sphinx-authbuild webserver does not like it and answers 403s

It could be argued that this should be brought to `sphinx-autobuild`'s attention as well. Note that RTD and GH-pages haven't stumbled over this extra `/`.

**Comment by hugoShaka** [comment]
> Have you tried that solution about removing the / in getAbsoluteUrl? I'm concerned that might lead to a regression.
I am not able to build the front because of corporate proxies blocking npm repos 🤷 . I will try again later today or tomorrow. Once this is properly tested I'll send a PR.

> There are other ways to host a webserver that "watches" for changes about a certain path.
Sure, this is not blocking for us, especially as we're serving the final website using nginx which handles such paths properly. But from a user point of view it feels a bit weird to change our local toolchain because of a theme change.

> It could be argued that this should be brought to sphinx-autobuild's attention as well. Note that RTD and GH-pages haven't stumbled over this extra /.
Makes sense, I opened an issue in sphinx-autobuild.

**Comment by jbms** [comment]
Yes, I think `config.base` always ends in `/` so the extra `/` added by `sphinx_search.ts` is redundant.  `sphinx_search.ts` is not from the upstream mkdocs-material package so there are no reservations in fixing the problem there.

**Comment by 2bndy5** [comment]
@hugoShaka you could make a commit on your fork (make sure that GH actions are enabled beforehand) and install the built wheel from the workflow's artifacts - no npm needed.
